.github/workflows/test-integration-postgres-redis-rabbit-matrix.yml

```diff
@@ -0,0 +1,171 @@
+name: Reusable workflow for running postgres with rabbit and redis integration tests in a Github Matrix
+
+on:
+  workflow_call:
+    secrets:
+      token:
+        description: 'The GitHub/npm token'
+        required: true
+    inputs:
+      test_command:
+        description: 'The command to run the integration tests'
+        default: 'npm run test:ci'
+        required: false
+        type: string
+      setup_global_postgres:
+        description: 'Setup Postgres @openphone/database repo'
+        default: true
+        required: false
+        type: boolean
+      global_postgres_ref:
+        description: 'Branch of Postgres @openphone/database repo to use'
+        default: 'master'
+        required: false
+        type: string
+      migration_command:
+        description: 'The command to run the Postgres migrations'
+        default: 'npm run db:up'
+        required: false
+        type: string
+      skip_migration:
+        description: 'Whether or not the migrations should be skipped. Useful for when you want to run specific migration tests'
+        default: false
+        required: false
+        type: boolean
+      postgres_version:
+        description: 'The version of Postgres to use'
+        default: '13'
+        required: false
+        type: string
+      postgres_user:
+        description: 'The Postgres user'
+        default: 'openphone'
+        required: false
+        type: string
+      postgres_password:
+        description: 'The Postgres password'
+        default: 'docker'
+        required: false
+        type: string
+      postgres_host:
+        description: 'The Postgres host'
+        default: 'localhost'
+        required: false
+        type: string
+      postgres_database:
+        description: 'The Postgres database'
+        default: 'openphone'
+        required: false
+        type: string
+      working-directory:
+        description: 'The path to the directory to run commands'
+        required: false
+        type: string
+        default: ./
+      timeout-minutes:
+        description: 'The max time to allow the action to run'
+        required: false
+        type: number
+        default: 10
+      shards:
+        description: 'The number of shards to use for Jest tests. Defaults to 1'
+        required: false
+        type: number
+        default: '1'
+
+jobs:
+  setup-matrix:
+    name: Generate test matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix-json: ${{ steps.setup-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup matrix
+        id: setup-matrix
+        uses: OpenPhone/gha/.github/actions/setup-matrix@zach-voice-8352-leverage-github-action-matrix-for-jest-tests
```

Comment:
I have this branch referencing itself so that I can still run the workflow. Once this PR gets merged, I will raise an immediate follow up to point this to main.

Reply:
These changes should be merged to `main` and then a release PR should be created to merge into `v5`. This should reference `OpenPhone/gha/.github/actions/setup-matrix@v5`